tests/cpptests/test_cpp_query.cpp

```diff
@@ -530,6 +530,39 @@ TEST_F(QueryTest, testParser_v2) {
   assertInvalidQuery("(hello world => [KNN 10 @vec_field $BLOB]) @title:other", ctx);
   assertInvalidQuery("hello world => [KNN 10 @vec_field $BLOB] OR other => [KNN 10 @vec_field $BLOB]", ctx);
 
+  // Test range queries
+  assertValidQuery("@v:[VECTOR_RANGE 0.01 $BLOB]", ctx);
+  assertValidQuery("@v:[vector_range 0.01 $BLOB]", ctx);
+  assertValidQuery("@v:[vEcToR_RaNgE 0.01 $BLOB]", ctx);
+  assertValidQuery("@v:[VECTOR_RANGE 2 $BLOB]", ctx);
+  assertValidQuery("@v:[VECTOR_RANGE $radius $BLOB]", ctx);
+  assertValidQuery("@v:[VECTOR_RANGE 2e-2 $BLOB]", ctx);
+  assertValidQuery("@v:[VECTOR_RANGE 2E-2 $BLOB]", ctx);
```

Comment:
Are we checking for negative values? When are we failing if so?

Reply:
That's a good point. Parser doesn't do the validation, I think it should be the iterator's job. Also, in vecsim lib we throw a runtime error if we get a negative range